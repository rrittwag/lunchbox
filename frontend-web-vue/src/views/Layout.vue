<template>
  <div class="relative flex min-h-screen w-full max-w-4md flex-col font-sans antialiased">
    <Header />
    <Content class="grow" />
  </div>
</template>

<script setup lang="ts">
import Content from '@/views/layout/Content.vue'
import Header from '@/views/layout/Header.vue'
import { useTheme } from '@/store/theme'

useTheme()
</script>
