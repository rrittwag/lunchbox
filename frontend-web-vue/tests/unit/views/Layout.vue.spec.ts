import Layout from '@/views/Layout.vue'
import { shallowMount } from '@vue/test-utils'
import { createTestingPinia } from '@tests/unit/test-utils'

describe('Layout', () => {
  test('renders snapshot', () => {
    const pinia = createTestingPinia()

    const wrapper = shallowMount(Layout, {
      global: { plugins: [pinia] },
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
