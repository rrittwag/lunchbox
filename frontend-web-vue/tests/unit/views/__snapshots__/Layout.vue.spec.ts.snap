// Vitest Snapshot v1

exports[`Layout renders snapshot 1`] = `
<div
  class="flex flex-col w-full max-w-4md min-h-screen font-sans antialiased relative"
>
  <header-stub />
  <content-stub
    class="grow"
  />
</div>
`;
